impl/src/main/java/org/ehcache/impl/persistence/FileBasedStateRepository.java

```diff
@@ -96,7 +94,7 @@ private void saveMaps() throws IOException {
   private String createFileName(Map.Entry<String, Tuple> entry) {return HOLDER_FILE_PREFIX + entry.getValue().index + "-" + safeIdentifier(entry.getKey(), false) + HOLDER_FILE_SUFFIX;}
 
   @Override
-  public <K extends Serializable, V extends Serializable> StateHolder<K, V> getPersistentStateHolder(String name, Class<K> keyClass, Class<V> valueClass) {
+  public <K extends Serializable, V extends Serializable> StateHolder<K, V> getPersistentStateHolder(String name, Class<K> keyClass, Class<V> valueClass, Iterable<Class> whiteListedClasses) {
```

Comment:
Should this be whitelisting?

Reply:
Yes!

Reply:
change parameter name to "whitelisting" instead of "whiteListedClasses" is what you are looking at ?

Reply:
No - I mean that there is no enforcement of the whitelist here.

Reply:
Had a chat with @rahul-mittal and because this class eagerly loads the saved data, it stays as-is for now.